### PR TITLE
Remember current project while navigating.

### PR DESCRIPTION
**Pull request by jamesmcclain**
It is now possible to navigate away from the /project page using the UI arrow buttons then return (using the UI arrow buttons) back to the project on which you have been working.

Connects #690

**To Test**
   * Do some activities which take you to the `/project` page.
   * Navigate away from the project page using the UI arrow buttons (back to the `/analyze` page or back to the root page), then back to the `/project` page.
   * You should find yourself back on the project that you were on when you left.

**Caveats**
   * I originally wanted to be able to take you back to the scenario that you had been working on, but unfortunately,this seems difficult.  One cannot set the scenario unless the DOM is in particular state: I am not able to give a precise description of this state and moreover I do not know how to wait for such a state without polling, which is a no-no (the usual expedient of releasing a lock within the `onShow`, `onRender` and `onAttach` functions of the views which create the DOM elements which I believe are required did not work).

### CODE REVIEW

**Comment by kdeloach** [comment]
One problem I found is that if you save a project, then analyze a new area, the previous project is loaded  instead of a blank project.

**Comment by jamesmcclain** [comment]
> One problem I found is that if you save a project, then analyze a new area, the previous project is loaded instead of a blank project.

Thanks for detecting this.  I have addressed this and the other things you mentioned.  Please have another look when you get a chance.

**Comment by lliss** [comment]
It seems that this feature negates the need for single project mode.
That was designed to ensure that a student who returns to the site outside of the ITSI portal would be able to continue work on the same project. It feels a lot like this handles the same cases. If this is true the singleProjectMode logic could be removed and some of the logic for loading the project page might become simpler.

**Comment by kdeloach** [comment]
+1 tested

**Comment by caseypt** [comment]
@lliss Should we create an issue to consolidate/remove "Single Project Mode" with this code?

**Comment by lliss** [comment]
Yes. This PR covers most but not all of the features of "Single Project Mode" notably the idea that you can move around without destroying the current project. There are some differences that we should resolve. For instance "Single Project Mode" will create a project for the user as soon as they visit the home page. This is then passed back to ITSI portal as the "submission" for the assignment.

The biggest difference is that new projects are not created in Single Project Mode. So when you redraw an area of interest, the current project data is wiped out and the new AOI data is added to it. Therefore you are always on the same project. The idea being that a student can rework the submission a few times but the submission id that the teacher will be aware of remains the same.

So in short, there is overlap and we can consolidate features but this doesn't totally superseded the need for single project mode.
